dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/SecurityUtilsService.kt

```diff
@@ -79,4 +89,38 @@ class SecurityUtilsService(
                 (currentRequestStatus == RequestStatus.Answered && requestStatusToPatch == RequestStatus.Open)
             )
     }
+
+    /**
+     * Returns true if the user is member of the company
+     * @param companyId dataland companyId
+     */
+    @Transactional
+    fun isUserMemberOfTheCompany(
+        companyId: UUID,
+    ): Boolean {
+        val userId = SecurityContextHolder.getContext().authentication.name ?: return false
+        return companyRoleAssignmentRepository.findByCompanyIdAndUserId(companyId.toString(), userId).isNotEmpty()
+    }
+
+    /**
+     * Returns true if the user has the rights to add/remove the companyRole
+     * @param companyId dataland companyId
+     * @param companyRoleToModify the companyRole to add/remove
+     */
+    @Transactional
+    fun hasUserPermissionToModifyTheCompanyRole(
+        companyId: UUID,
+        companyRoleToModify: CompanyRole,
+    ): Boolean {
+        val userId = SecurityContextHolder.getContext().authentication.name ?: return false
+        val userCompanyRoles =
+            companyRoleAssignmentRepository.findByCompanyIdAndUserId(companyId.toString(), userId)
+        var hasUserPermission = false
+        userCompanyRoles.forEach {
+            if (roleModificationPermissionsMap[it.companyRole]?.contains(companyRoleToModify) == true) {
+                hasUserPermission = true
+            }
+        }
```

Comment:
Grundsätzlich top. Nur ein Mini-Vorschlag, der hier jedoch wirklich kaum was ausmacht an der Performance, aber den Code ein wenig verkürzt:

Du könntest sobald die if-Bedingung erfüllt ist direkt `return true` schreiben.  Dadurch unterbrichst du die forEach-Loop, weil dich interessiert ja ab dann nicht mehr, ob der user auch noch eine andere Rolle haben könnte, die die modification ebenfalls erlaubt. Also vermeidest du weitere Iterationen.

Ebenso könntest du dann ganz unten in der Funktion einfach "return false" schreiben, weil wenn die Funktion bis dahin durchgelaufen ist, dann besitzt der user offenbar keine modification-Rechte.

Dadurch wird die Variable `hasUserPermission` glaube ich auch obsolet.

Reply:
hatte ich auch erst, aber dann hatte sonar genervt wegen 3 return statements.

Reply:
Hat sich erledigt durch den neuen code